// srod-v2/srod-v2.js
/**
 * warning: iw as in drug whilem kaning this pakge so yeah..
 * most of mtiem i am druk but who cares
 */

module.exports = {
    Version: require("./package.json").version,
    Awooify: require("./Src/Main.js").Awooify,
    Baguette: require("./Src/Main.js").Baguette,
    BlushFact: require("./Src/Main.js").BlushFact,
    ChangeMyMind: require("./Src/Main.js").ChangeMyMind,
    Chat: require("./Src/Main.js").Chat,
    Clyde: require("./Src/Main.js").Clyde,
    DiscordJSDocs: require("./Src/Main.js").DiscordJSDocs,
    DiscordSpoiler: require("./Src/Main.js").DiscordSpoiler,
    Gay: require("./Src/Main.js").Gay,
    GetAdvice: require("./Src/Main.js").GetAdvice,
    GetAnimalImage: require("./Src/Main.js").GetAnimalImage,
    GetAnimeImage: require("./Src/Main.js").GetAnimeImage,
    GetFact: require("./Src/Main.js").GetFact,
    GetJoke: require("./Src/Main.js").GetJoke,
    GetMeme: require("./Src/Main.js").GetMeme,
    GetWhy: require("./Src/Main.js").GetWhy,
    Glass: require("./Src/Main.js").Glass,
    HasteBin: require("./Src/Main.js").HasteBin,
    KannaGen: require("./Src/Main.js").KannaGen,
    Lmgtfy: require("./Src/Main.js").Lmgtfy,
    MCUserID: require("./Src/Main.js").MCUserID,
    PornhubComment: require("./Src/Main.js").PornhubComment,
    RandomString: require("./Src/Main.js").RandomString,
    Threats: require("./Src/Main.js").Threats,
    Triggered: require("./Src/Main.js").Triggered,
    TrumpTweet: require("./Src/Main.js").TrumpTweet,
    Tweet: require("./Src/Main.js").Tweet,
    Wasted: require("./Src/Main.js").Wasted,
    WhoWouldWin: require("./Src/Main.js").WhoWouldWin,
    YoutubeComment: require("./Src/Main.js").YoutubeComment
};
