// srod-v2/srod-v2.js
/**
 * warning: iw as in drug whilem kaning this pakge so yeah..
 * most of mtiem i am druk but who cares
 */

module.exports = {
    Version: require("./package.json").version,
    Awooify: require("./Src/Awooify.js"),
    Baguette: require("./Src/Baguette.js"),
    BlushFact: require("./Src/BlushFact.js"),
    ChangeMyMind: require("./Src/ChangeMyMind.js"),
    Chat: require("./Src/Chat.js"),
    Clyde: require("./Src/Clyde.js"),
    DiscordJSDocs: require("./Src/DiscordJSDocs.js"),
    DiscordSpoiler: require("./Src/DiscordSpoiler.js"),
    Gay: require("./Src/Gay.js"),
    GetAdvice: require("./Src/GetAdvice.js"),
    GetAnimalImage: require("./Src/GetAnimalImage.js"),
    GetAnimeImage: require("./Src/GetAnimeImage.js"),
    GetFact: require("./Src/GetFact.js"),
    GetJoke: require("./Src/GetJoke.js"),
    GetMeme: require("./Src/GetMeme.js"),
    GetTopic: require("./Src/GetTopic.js"),
    GetWhy: require("./Src/GetWhy.js"),
    GetWord: require("./Src/GetWord.js"),
    Glass: require("./Src/Glass.js"),
    HasteBin: require("./Src/HasteBin.js"),
    KannaGen: require("./Src/KannaGen.js"),
    Lmgtfy: require("./Src/Lmgtfy.js"),
    MCUserID: require("./Src/MCUserID.js"),
    PornhubComment: require("./Src/PornhubComment.js"),
    RandomString: require("./Src/RandomString.js"),
    Threats: require("./Src/Threats.js"),
    Translate: require("./Src/Translate.js"),
    Triggered: require("./Src/Triggered.js"),
    TrumpTweet: require("./Src/TrumpTweet.js"),
    Tweet: require("./Src/Tweet.js"),
    Wasted: require("./Src/Wasted.js"),
    WhoWouldWin: require("./Src/WhoWouldWin.js"),
    YoutubeComment: require("./Src/YoutubeComment.js")
};
