// sakuranpm/src/index.js
const fetch = require('node-fetch');
let baseUrl = "https://api.skapi.ga/api/img";//Sakura Kinomoto Bot Api
module.exports = {
  /* Gifs */
  kiss: async function () {
    let api = await fetch(`${baseUrl}/kiss`).then(res => res.json());
    return api.url;
  },
  vomit: async function () {
    let api = await fetch(`${baseUrl}/vomit`).then(res => res.json());
    return api.url;
  },
  hug: async function () {
    let api = await fetch(`${baseUrl}/hug`).then(res => res.json());
    return api.url;
  },
  pat: async function () {
    let api = await fetch(`${baseUrl}/pat`).then(res => res.json());
    return api.url;
  },
  cry: async function () {
    let api = await fetch(`${baseUrl}/cry`).then(res => res.json());
    return api.url;
  },
  hi: async function () {
    let api = await fetch(`${baseUrl}/hi`).then(res => res.json());
    return api.url;
  },
  bye: async function () {
    let api = await fetch(`${baseUrl}/bye`).then(res => res.json());
    return api.url;
  },
  nope: async function () {
    let api = await fetch(`${baseUrl}/nope`).then(res => res.json());
    return api.url;
  },
  bored: async function () {
  	let api = await fetch(`${baseUrl}/bored`).then(res => res.json());
  	return api.url;
  },
  tickle: async function () {
    let api = await fetch(`${baseUrl}/tickle`).then(res => res.json());
    return api.url;
  },
  scared: async function () {
    let api = await fetch(`${baseUrl}/scared`).then(res => res.json());
    return api.url;
  },
  blush: async function () {
    let api = await fetch(`${baseUrl}/blush`).then(res => res.json());
    return api.url;
  },
  claps: async function () {
    let api = await fetch(`${baseUrl}/claps`).then(res => res.json());
    return api.url;
  },
  kill: async function () {
    let api = await fetch(`${baseUrl}/kill`).then(res => res.json());
    return api.url;
  },
  sad: async function () {
    let api = await fetch(`${baseUrl}/sad`).then(res => res.json());
    return api.url;
  },        


  /* Images */  
  meme: async function () {
    let api = await fetch(`${baseUrl}/meme`).then(res => res.json());
    return api.url;
  },
  shitpost: async function () {
    let api = await fetch(`${baseUrl}/shitpost`).then(res => res.json());
    return api.url;
  },
  neko: async function () {
    let api = await fetch(`${baseUrl}/neko`).then(res => res.json());
    return api.url;
  }     
}
