// nekobot-api/dist/utils/Request.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Error_1 = __importDefault(require("./Error"));
const https_1 = require("https");
class Request {
    constructor(client) {
        this.client = client;
    }
    get(endpoint, { queries, headers = {} }) {
        return new Promise((resolve, reject) => {
            this.
                _send("GET", endpoint, { queries, headers })
                .then(x => resolve(x.body.message))
                .catch(reject);
        });
    }
    _send(method, endpoint, { queries, headers = {} }) {
        const pkg = require("../../package.json");
        return new Promise((resolve, reject) => {
            const req = https_1.request(`${this.client.baseURL}/${endpoint}?${this.encodeSearchParams(queries)}`, {
                method,
                headers: Object.assign({ "User-Agent": `${pkg.name}/${pkg.version} (${pkg.repository.url})` }, headers)
            }, response => {
                let raw = "";
                response
                    .on("data", chunk => {
                    raw += chunk;
                })
                    .on("end", () => {
                    const res = {
                        statusCode: response.statusCode,
                        body: JSON.parse(raw)
                    };
                    if (!(res.statusCode >= 200 && res.statusCode < 300)) {
                        reject(new Error_1.default(res.body.message, res.statusCode));
                    }
                    else {
                        resolve(res);
                    }
                })
                    .on("error", reject);
            });
            req.end();
        });
    }
    encodeSearchParams(queries) {
        return Object.entries(queries)
            .filter(x => Boolean(x[1]))
            .map(x => `${x[0]}=${encodeURIComponent(x[1])}`)
            .join("&");
    }
}
exports.default = Request;
