// imdb-api/lib/interfaces.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.assertSearchResponse = exports.assertGetResponse = exports.assertEpisodeSeasonResponse = exports.isGame = exports.isEpisode = exports.isMovie = exports.isTvshow = exports.isError = void 0;
function isError(response) {
    return response.Response === "False";
}
exports.isError = isError;
function isTvshow(response) {
    return response.Type === "series";
}
exports.isTvshow = isTvshow;
function isMovie(response) {
    return response.Type === "movie";
}
exports.isMovie = isMovie;
function isEpisode(response) {
    return response.Type === "episode";
}
exports.isEpisode = isEpisode;
function isGame(response) {
    return response.Type === "game";
}
exports.isGame = isGame;
function assertEpisodeSeasonResponse(response) {
    if (!Array.isArray(response)) {
        return false;
    }
    for (var _i = 0, response_1 = response; _i < response_1.length; _i++) {
        var res = response_1[_i];
        if (typeof res !== "object") {
            return false;
        }
        if ("Response" in res) {
            if (res.Response === "False") {
                return true;
            }
            if ("Season" in res && "totalSeasons" in res) {
                return true;
            }
        }
    }
    return false;
}
exports.assertEpisodeSeasonResponse = assertEpisodeSeasonResponse;
function assertGetResponse(response) {
    if (typeof response !== "object") {
        return false;
    }
    var res = response;
    if (res === null) {
        return false;
    }
    if ("Response" in res) {
        return true;
    }
    return false;
}
exports.assertGetResponse = assertGetResponse;
function assertSearchResponse(response) {
    if (typeof response !== "object") {
        return false;
    }
    var res = response;
    if (res === null) {
        return false;
    }
    if ("Response" in res) {
        if (res.Response === "False") {
            return true;
        }
    }
    if ("Search" in res) {
        return true;
    }
    return false;
}
exports.assertSearchResponse = assertSearchResponse;
